src/main/com/example011/radek/v1/Mines.java

```diff
@@ -0,0 +1,138 @@
+package main.com.example011.radek.v1;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Scanner;
+
+public class Mines {
+
+    protected int playgroundSize;
+
+    protected int numberOfMines;
+
+    protected String[][] playground;
+
+    public Mines(int size, int mines) {
+
+        if (checkPlaygroundSize(size)) {
+            setPlaygroundSize(size);
+            setNumberOfMines(mines);
+        } else {
+            System.out.println("Bad size of playground, size must be bigger then 5!");
+            System.exit(0);
+        }
+    }
+
+    public void fillPlaygroundWithMines() throws NoSuchAlgorithmException {
+
+        playground = new String[playgroundSize][playgroundSize];
+        SecureRandom random = SecureRandom.getInstanceStrong();
+        int placedMines = 0;
+        while (placedMines != getNumberOfMines() || placedMines == 0) {
+            int generatedRowNumber = random.nextInt(playgroundSize);
+            int generatedColumnNumber = random.nextInt(playgroundSize);
+            if (playground[generatedRowNumber][generatedColumnNumber] == null) {
+                playground[generatedRowNumber][generatedColumnNumber] = StringConstant.MINE.getConstant();
+                placedMines++;
+            }
+        }
+    }
+
+    public void printPlayground(boolean showMines) {
+
+        System.out.println("Generating playground...");
+        StringBuilder playingField = new StringBuilder(playgroundSize * playgroundSize * 3);
+        for (int i = 0; i < playgroundSize; i++) {
+            for (int j = 0; j < playgroundSize; j++) {
+                if (playground[i][j] != null && ((showMines && playground[i][j].equals("x")) || playground[i][j].equals("o"))) {
```

Comment:
uz mame enum pro tyto stringy

Reply:
Opraveno, podmínku jsem rozepsal do metod zatím, už to bylo dost dlouhé

Reply:
celkove uz mnohem lepsi, na juniora dobre

Reply:
Musí to být výborné :D Musím přidat tempo